File: JavaScript/Stack/Stack.js

    // Time Complexity
    // ---------------
    // Push        O(1)
    // Pop         O(1)
    // is_empty    O(1)
    // is_full     O(1)
    // peek        O(1)
    // size        O(1)




class Stack {
    constructor(capacity = 5) {
        this.items = [];
        this.capacity = capacity;
    }

    is_empty(){
        return this.items.length <= 0;
     }

     is_full(){
        return this.items.length == 5;
     }
    
    
    Push(element) {
        if (this.is_full()) console.log("Stack Overflow!");
        else this.items.push(element);
    }
    
    
    Pop() {
        if (this.is_empty()) console.log("Stack Underflow!")
        else {return this.items.pop();}
        
    }
    
    peek() {
        return this.items[this.items.length - 1];
    }
    
    

    size(){
        return this.items.length;
    }
 
}

//main
let stack = new Stack();
stack.Pop()
stack.Push(10);
stack.Push(20);
stack.Push(30);
stack.Push(40);
stack.Push(50);
stack.Push(60);

console.log(stack.size());
stack.Pop()
stack.Pop()
console.log(stack.peek())

File: Stack.py
"""
    Time Complexity
    ---------------
    push        O(1)
    pop         O(1)
    is_empty    O(1)
    is_full     O(1)
    peek        O(1)
    size        O(1)
"""


class Stack(object):
    def __init__(self, capacity=5):
        self.stack = []
        self.capacity = capacity

    def is_empty(self):
        return len(self.stack) <= 0

    def is_full(self):
        return len(self.stack) >= self.capacity

    def push(self, item):
        if self.is_full():
            print("Stack Overflow!")
        else:
            self.stack.insert(0, item)

    def pop(self):
        if self.is_empty():
            print("Stack Underflow")
        else:
            self.stack.pop(0)

    def peek(self):
        return self.stack[0]

    def size(self):
        return len(self.stack)


#if __name__ == "__main__":
   # st = Stack()
 #   st.pop()            # Checking underflow condition
   # st.push(10)
   # st.push(20)
   # st.push(30)
 #   st.push(40)
  #  st.push(50)
   # st.push(60)         # Checking overflow condition
   # print(st.size())
   # st.pop()
   # st.pop()
   # print(st.peek())
